ciao-controller/configuration.go

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/01org/ciao/ciao-controller/types"
+	"github.com/01org/ciao/payloads"
+	"github.com/01org/ciao/service"
+	"github.com/01org/ciao/ssntp"
+	"github.com/golang/glog"
+	"gopkg.in/yaml.v2"
+)
+
+func (c *controller) UpdateClusterConfig(ctx context.Context, newConf types.ConfigRequest, token string) error {
+	var pyld payloads.Configure
+	tenantID, err := service.GetTenantID(ctx)
+	if err != nil {
+		return errors.New("Unable to retrieve tenant ID")
+	}
+	if newConf.Element == "" {
+		return errors.New("empty Element")
+	}
+	if newConf.Value == "" {
+		return errors.New("empty Value")
+	}
+	if validConfigElement(newConf.Element) == false {
+		return errors.New("invalid Value")
+	}
+	if validConfigValue(newConf.Value, newConf.Element) == false {
+		return errors.New("invalid Element")
+	}
+	pyld, err = c.client.ssntpClient().ClusterConfiguration()
+	if err != nil {
+		return err
+	}
+	err = testNewConfig(newConf, token, tenantID, pyld)
+	if err != nil {
+		return err
+	}
+	err = updateConfPayload(newConf, &pyld)
+	if err != nil {
+		return err
+	}
+	newConfPyld, err := yaml.Marshal(pyld)
+	if err != nil {
+		return err
+	}
+	c.client.ssntpClient().SendCommand(ssntp.CONFIGURE, newConfPyld)
+	return nil
+}
+
+func (c *controller) ShowClusterConfig() (string, error) {
+	pyld, err := c.client.ssntpClient().ClusterConfiguration()
+	if err != nil {
+		return "", err
+	}
+	// no need to share sensitive information
+	pyld.Configure.Controller.IdentityPassword = "*****"
```

Comment:
Why bother transmitting this information at all?  If we set it null can we get the yaml parser to simply exclude these fields?